### DISCUSSION TITLE
Incomplete button switches on Sideband

### DISCUSSION THREAD

**Discussion by hardBSDk** [markqvist/Reticulum#136] [Help & Questions]
On Preferences section, the switch buttons are incomplete, they just work when clicking on the dot inside the button, not any area of the button.

**Comment by markqvist** [comment]
Yes, I know of this. It is super annoying, it is an effect of the MDSwitch class in KivyMD, and it is apparently designed like that. So far I have not found a way to modify the behaviour, but have not looked much into it yet.

The code is [here](https://github.com/markqvist/Sideband/blob/1603fd4a0f8108651cbfffca5e69a1a542f5a935/sbapp/kivymd/uix/selectioncontrol/selectioncontrol.py#L107) and [here](https://github.com/markqvist/Sideband/blob/1603fd4a0f8108651cbfffca5e69a1a542f5a935/sbapp/kivymd/uix/selectioncontrol/selectioncontrol.py#L445), if anyone wants to have a look at how to fix it :)

**Reply by markqvist** [reply]
It is in KivyMD, not Kivy itself ;) kivymd is built on top of kivy. A bit confusing, I know.

**Reply by markqvist** [reply]
And I am not really sure if it is meant to be that way. Either way, we should be able to just modify the class locally for the intended effect.

**Reply by hardBSDk** [reply]
Sorry for my mistake, I tough it belongs to Kivy umbrella.

Reported correctly.

https://discord.com/channels/566880874789076992/1035374446518747136

**Reply by hardBSDk** [reply]
@markqvist Where's the code of Preferences switch buttons? IDK.

**Reply by markqvist** [reply]
The layouts are defined here: https://github.com/markqvist/Sideband/blob/main/sbapp/ui/layouts.py#L1027

They just use the base MDSwitch class, so if it can be subclassed into something that behaves correctly, we can just use that instead. Or we can just patch the MDSwitch class, since Sideband has a completely local copy of KivyMD. I am manually tracking changes on KivyMD since there is so much development going on there, that stuff often breaks, and everytime you update, something needs fixing ;)

**Reply by hardBSDk** [reply]
@markqvist Which code represents the button switches?

![image](https://user-images.githubusercontent.com/55782009/198839475-da16dfb6-7fff-4100-b22b-530290d547b0.png)

**Reply by markqvist** [reply]
They want just a small runnable code sketch that demonstrates the issue, so just like a few lines that creates a switch, and then a description like "When clicking the switch, it only triggers if the click lands exactly on the circle part of the switch. I don't have time to do it right now, though.
